fpmsyncd/fpmlink.cpp

```diff
@@ -277,3 +279,31 @@ void FpmLink::processFpmMessage(fpm_msg_hdr_t* hdr)
         nlmsg_free(msg);
     }
 }
+
+bool FpmLink::send(nlmsghdr* nl_hdr)
+{
+    fpm_msg_hdr_t hdr;
+
+    size_t len = fpm_msg_align(sizeof(hdr) + nl_hdr->nlmsg_len);
+
+    hdr.version = FPM_PROTO_VERSION;
+    hdr.msg_type = FPM_MSG_TYPE_NETLINK;
+    hdr.msg_len = htons(static_cast<uint16_t>(len));
+
+    memcpy(m_sendBuffer, &hdr, sizeof(hdr));
+    memcpy(m_sendBuffer + sizeof(hdr), nl_hdr, nl_hdr->nlmsg_len);
```

Comment:
Do safe copy (with original buffer size check)

Reply:
It's str* functions that have a safe variants. I just put a check before that len should be less then or eq m_bufSize.